build.gradle

```diff
@@ -365,6 +405,39 @@ subprojects {
         sign configurations.archives
     }
 
+    if (project.hasProperty("minCoverageRatio")) {
+        def minCoverageRatio = (project.findProperty("minCoverageRatio") ?: "0.75") as double
+
+        // Configure the JaCoCo test coverage verification
+        jacocoTestCoverageVerification {
+            violationRules {
+                rule {
+                    limit {
+                        minimum = minCoverageRatio
+                    }
+                }
+            }
+            afterEvaluate {
+                // Ignore Scala classes.
+                classDirectories = files(classDirectories.files.collect { classDirectory ->
```

Comment:
This was needed twice?

Reply:
Unfortunately, yes. In one case, we are configuring the reporting task to ignore Scala sources. In the other case, we are configuring the verification task to ignore Scala sources. JaCoCo itself extends test tasks to generate the `.exec` files with all of the data. Both the report and the verification use those files to do their thing. The key there is that the verification task does not use the reports from the report task, so we have to ignore those classes in both places.

I mean, unless we don't care about reports. But it costs us nothing, so why not keep it configured?

Reply:
you've created two sources of truth. I don't mind it being there for the reports, i just wish it were easier to have it written out once, in case we end up having to change it.

Reply:
I recognize that, and did not really want to do it... which is why I extracted the bulk of this out into a method. However, that seems to be the way the plugin works, so I'm not sure how to get around that, especially since in one case configuration is conditional (i.e., executed based on the presence of the `minCoverageRatio` Project property).